src/dotnet/commands/dotnet-restore/Program.cs

```diff
@@ -27,8 +27,16 @@ public static int Run(string[] args)
 
             app.OnExecute(() =>
             {
+                const string nugetXProjWriteTargets = "NUGET_XPROJ_WRITE_TARGETS";
+                bool setXProjWriteTargets = Environment.GetEnvironmentVariable(nugetXProjWriteTargets) == null;
+
                 try
                 {
+                    if (setXProjWriteTargets)
+                    {
```

Comment:
If the environment variable is already set, set it again, and then remove whatever value it had originally..

Why do we need to do this?